src/coprocessor/codec/convert.rs

```diff
@@ -261,6 +300,81 @@ pub fn convert_float_to_uint(ctx: &mut EvalContext, fval: f64, tp: FieldTypeTp)
     Ok(val as u64)
 }
 
+/// Converts a byte arrays to an u64 in best effort.
+#[inline]
+pub fn convert_bytes_to_uint(ctx: &mut EvalContext, bytes: &[u8], tp: FieldTypeTp) -> Result<u64> {
+    let s = get_valid_utf8_prefix(ctx, bytes)?;
+    let vs = get_valid_int_prefix(ctx, s)?;
+    let val = vs.parse::<u64>();
+    match val {
+        Ok(val) => convert_uint_to_uint(ctx, val, tp),
+        Err(_) => {
+            ctx.handle_overflow(Error::overflow("BIGINT UNSIGNED", &vs))?;
+            let val = integer_unsigned_upper_bound(tp);
+            Ok(val)
+        }
+    }
+}
+
+#[inline]
+fn decimal_as_u64(ctx: &mut EvalContext, dec: Decimal, tp: FieldTypeTp) -> Result<u64> {
+    let val = match dec.as_u64() {
+        Res::Ok(val) => val,
+        Res::Overflow(val) => {
+            ctx.handle_overflow(Error::overflow("DECIMAL", &dec.to_string()))?;
+            val
```

Comment:
the `dec.to_string()` is private function, why it can be called?

Reply:
The `to_string` is come from `std::convert::ToString` trait.